### PR TITLE
Add the registration bean for an annotated service object

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

We forgot to add a registration bean for an annotated service object.

Modifications:

- Add AnnotatedServiceRegistrationBean

Result:

A user can add an annotated service object to a Server without using the
configurator.

### CODE REVIEW

File: spring-boot/autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java [hunk 1]
```diff
@@ -149,6 +151,18 @@ public Server armeriaServer(
...
+        annotatedServiceRegistrationBeans.ifPresent(beans -> beans.forEach(bean -> {
+            final Function<Service<HttpRequest, HttpResponse>,
+                           ? extends Service<HttpRequest, HttpResponse>> decorator;
+            if (armeriaSettings.isEnableDropwizardMetrics()) {
+                decorator = DropwizardMetricCollectingService.newDecorator(
+                        metricRegistry, serviceMetricName(bean.getServiceName()));
+            } else {
+                decorator = Function.identity();
+            }
+            server.annotatedService(bean.getPathPrefix(), bean.getService(), decorator);
```
Comment (krrrr38) [inline spring-boot/autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java]: could you add `decorator` into `AnnotattedServiceRegistrationBean`?

Comment (trustin) [inline spring-boot/autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java]: Good idea. Done.

**Comment by trustin** [comment]
@imasahiro @anuraaga It seems like we are mixing `@Nonnull` and `@NotNull` in the bean implementations. Which should I use?

File: spring-boot/autoconfigure/src/main/java/com/linecorp/armeria/spring/ThriftServiceRegistrationBean.java [hunk 2]
```diff
@@ -17,12 +17,16 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.function.Function;
 
+import javax.annotation.Nonnull;
 import javax.validation.constraints.NotNull;
```
Comment (anuraaga) [inline spring-boot/autoconfigure/src/main/java/com/linecorp/armeria/spring/ThriftServiceRegistrationBean.java]: Java has way too many nullability annotations that mean the same thing... This one under validation triggers spring's checker and is appropriate for beans I think.

Comment (trustin) [inline spring-boot/autoconfigure/src/main/java/com/linecorp/armeria/spring/ThriftServiceRegistrationBean.java]: Let me use `NotNull` then for the beans.

Comment (trustin) [inline spring-boot/autoconfigure/src/main/java/com/linecorp/armeria/spring/ThriftServiceRegistrationBean.java]: Thanks for the answer. :-)

**Comment by imasahiro** [comment]
Just realized that it woud be useful if `AnnotatedServiceRegistrationBean` supports `setConverters` method since we sometimes want to use a converter that does not have default constructor like this:
```java
@Named
public class MyStringConverter implements ResponseConverter {
    private final Clock clock;
    @Inject
    public MyStringConverter(Clock clock) {  this.clock = clock; }

    @Override
    public HttpResponse convert(Object resObj) throws Exception {...}
}
```

**Pull request merged**